os/mm/mm_heap/mm_heapinfo.c

```diff
@@ -385,13 +407,14 @@ void heapinfo_update_group_info(pid_t pid, int group, int type)
 		}
 		break;
 	case HEAPINFO_ADD_INFO:
-		if (group > heap->max_group) {
-			heap->max_group = group;
+		if (group > max_group) {
+			max_group = group;
 		}
 		for (info_idx = 0; info_idx < HEAPINFO_THREAD_NUM; info_idx++) {
 			if (group_info[info_idx].pid <= 0) {
 				group_info[info_idx].pid = pid;
 				group_info[info_idx].group = group;
+				tcb = sched_gettcb(pid);
```

Comment:
If this if condition at line 414 equals false, then in that case tcb would be unused & can throw error.
Is there a possibility of 
if (group_info[info_idx].pid <= 0)  
condition failing ???

Reply:
yes. it is checked before calling "heapinfo_update_group_info"